test/python/transpiler/test_sabre_layout.py

```diff
@@ -254,7 +254,7 @@ def test_layout_many_search_trials(self):
         self.assertIsInstance(res, QuantumCircuit)
         layout = res._layout.initial_layout
         self.assertEqual(
-            [layout[q] for q in qc.qubits], [22, 7, 2, 12, 1, 5, 14, 4, 11, 0, 16, 15, 3, 10]
+            [layout[q] for q in qc.qubits], [10, 16, 8, 4, 11, 20, 15, 19, 18, 7, 12, 1, 2, 0]
         )
 
 
```

Comment:
Do we know why this changed?  The actual barriers inserted before routing should have been the same, right, so it shouldn't have affected StochasticSwap?  It's not really a problem, I just don't understand why it's changed.

Reply:
My assumption was because the input circuit has barriers in it that this was a side effect of not running merge adjacent barriers part of the barrier insertion. (this only changed with https://github.com/Qiskit/qiskit/pull/10323/commits/29c121724881189287365c0c903c436d5c97d6e8 ). So, now we multiple barriers followed by an all qubit barrier instead of it all getting merged into a single barrier. But I didn't look too closely at the internals of what was causing sabre to diverge internally.

Reply:
Just a heads up I did get to the bottom of this, it was basically what I thought. I pushed up #11295 to fix the behavior of Sabre with regards to barriers and had to revert this change in that PR: https://github.com/Qiskit/qiskit/pull/11295/commits/93a32419778dc88423a0e4ee7d328cc431d71239